extensions/amp-app-banner/amp-app-banner.md

```diff
@@ -0,0 +1,158 @@
+
+The AMP runtime parses the meta tag content attribute on iOS extracting the App ID and `app-argument` (usually used for deep-linking URIs - app-protocols like `whatsapp://` or `medium://`). On Android, the AMP runtimes makes an XHR request to fetch the `manifest.json` file, and parses its content to extract `app_id` from `related_applications` and it calculates the app store URL as well as open-in-app URL:
+
+```
+android-app://${appId}/${protocol}/${pathname}
+```
+
+##Appearance Behavior
+amp-app-banner provides no default UI and leave the UI to the developer. The developer can build any kind of UI inside the banner and style it accordingly. The only UI element provided by AMP and can’t be overridden by publishers is the dismiss button that allows users to dismiss the banner.
+
+One required UI element is the `button[open-link]` button, which is the click target for the banner to install the app, or open the deep-link if the app is already installed.
+
+###OS/Browser/Viewer Dependencies
+
+Because native app banners currently are not shown in the viewer context, `<amp-app-banner>` is shown inside of the viewer when the appropriate markup is present. Because system-level app banners are already shown in Android/Chrome and iOS/Safari OS/Browser combinations, `<amp-app-banner>` is hidden in those contexts to avoid redundancy.
+
+<table>
+  <tr>
+    <td></tc>
+    <td>Android + Chrome</tc>
+    <td>iOS + Safari</tc>
+    <td>Other OS + browser</tc>
+  </tr>
+  <tr>
+    <td>In AMP viewer</tc>
+    <td>Show amp-app-banner</tc>
+    <td>Show amp-app-banner</tc>
+    <td>Show amp-app-banner</tc>
+  </tr>
+  <tr>
+    <td>Outside of AMP viewer</tc>
+    <td>Show system banner</tc>
+    <td>Show system banner</tc>
+    <td>Show amp-app-banner</tc>
+  </tr>
+</table>
+
+
+##Dismissal Persistence
+The banner currently will be displayed always unless it was dismissed. Once dismissed the banner will never be displayed on that domain unless user visits on a different browser or clears their local storage.
+
+The persistence is managed through AMP Local Storage by storing a boolean value with a key `amp-app-banner:<element.id>`.
+
+## Tags
+
+**At least one of `meta[name="apple-itunes-app"]` or `link[rel=manifest]`**
+* `<meta>` tag must have "name" attribute and "content" attribute
+* The value of the "content" attribute must contain "app-id=" 
+* `<link>` tag must have "rel='manifest'" attribute and value, as well as "href" attribute
+
+
+## Attributes
+
+
+### Attributes on `amp-app-banner`
+
+**id** (Required)
+
+To uniquely identify an amp-app-banner - for persistence logic
+
+**layout="nodisplay"** (Required)
+
+### Attributes on 'button' descendent element
+
+**open-link** (Required)
+
+
+## Additional Validations
+
+* Cannot have `<amp-ad>`, `<amp-sticky-ad>`, or `<amp-embed>` as a descendent
```

Comment:
Good call on amp-iframe—I'm not too worried about the other heavy components (video, commenting, etc), since they're less likely to be abused, but let me know if you're worried about any in particular